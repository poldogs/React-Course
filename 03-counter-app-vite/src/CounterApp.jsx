import { useState } from 'react';
import PropTypes from 'prop-types';

export const CounterApp = ({ value }) => {
    
    const [ counter, setCounter ] = useState( value );

    const handleAdd = () => { 
        // console.log(event)
        setCounter( counter + 1 );
        // setCounter( (c) => c + 1 )
    }

    const handleSubstract = () => setCounter( counter - 1 );
    const handleReset = () => setCounter( value );

    return (
        <>
            <h1>CounterApp</h1>
            <h2 id='counter'> { counter } </h2>

            <button onClick={ handleAdd } id='btn-add'> +1 </button>
            <button onClick={ handleSubstract } id='btn-substact'> -1 </button>
            <button aria-label="btn-reset" onClick={ handleReset }> Reset </button>
        </>
    );
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}


