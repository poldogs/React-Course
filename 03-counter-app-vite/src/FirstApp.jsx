import PropTypes from 'prop-types'


export const FirstApp = ({ title , subTitle, name }) => {

    const newMessage = {
        name: 'Leopoldo',
        age: 45
    }

  return (
    <>  {/* React.Fragment */}
        <h1>{ title }</h1>
        {<code>{JSON.stringify(newMessage)}</code>}
        <p>{ subTitle }</p>
        <p>{ name }</p>
    </>
  )
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string.isRequired
}

FirstApp.defaultProps = {
    name: 'React Default',
    subTitle: 'Subtitulo por defecto',
    title: 'Titulo por defecto'
}
