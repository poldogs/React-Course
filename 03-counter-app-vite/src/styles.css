html, body{
    background-color: #21232A;
    color: #61DAFB;
    font-family: 1,3rem;
    font-family: Helvetica, Arial, sans-serif;
    margin: 0;
    padding: 70px;
}

button{
    background-color: #61DAFB;
    border: none;
    border-radius: 5px;
    color: #21232A;
    cursor: pointer;
    font-size: 1rem;
    margin: 0 10px;
    padding: 10px 20px;
}