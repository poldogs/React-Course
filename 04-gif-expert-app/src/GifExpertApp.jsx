import React, { useState } from "react";
import { AddCategory } from "./components/AddCategory";

export const GifExpertApp = () => {

    const [categories, setcategories] = useState(['One Punch', 'Samurai X', 'Dragon Ball'])

    const handleAddCategory = () => {
        setcategories([...categories, inputValue]);
    }

    return (
        <>
            <h1>GifExpertApp</h1>

            {/* Input to add a new category */}
            <AddCategory setCategories={setcategories} />

            {/* List of categories */}
            <ol>
                {
                    categories.map( category => {
                        return <li key={category}>{category}</li>
                    })
                }
            </ol>
        </>
    );
    }