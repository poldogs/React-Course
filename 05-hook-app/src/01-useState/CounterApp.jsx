import { useState } from "react"


export const CounterApp = () => {

    const [counter, setCounter] = useState({
        counter1: 10,
        counter2: 20,
        counter3: 30
    })

  return (
    <>
        <h1>Counter:</h1>

        <h2>Counter 1: {counter.counter1}</h2>
        <h2>Counter 2: {counter.counter2}</h2>
        <h2>Counter 3: {counter.counter3}</h2>

        <button onClick={() => setCounter({ ...counter, counter1: counter.counter1 + 1 })}>+Counter1</button>
        <button onClick={() => setCounter({ ...counter, counter2: counter.counter2 + 1 })}>+Counter2</button>
        <button onClick={() => setCounter({ ...counter, counter3: counter.counter3 + 1 })}>+Counter3</button>
    </>
  )
}
