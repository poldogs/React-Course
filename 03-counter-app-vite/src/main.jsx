import React from 'react'
import ReactDOM from 'react-dom';
import { HelloWorldApp } from './HelloWorldApp';
import { FirstApp } from './FirstApp';
import { CounterApp } from './CounterApp';
import './styles.css';

ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        <HelloWorldApp />
        <FirstApp title="Probando"  subTitle="cosas"  />

        <CounterApp value={12}/>
    </React.StrictMode>
);