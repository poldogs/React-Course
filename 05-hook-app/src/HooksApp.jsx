import React from 'react'
import { CounterApp } from './01-useState/CounterApp'
import { CounterWithCustomHook } from './01-useState/CounterWithCustomHook'

export const HooksApp = () => {
  return (
    <div>
        <h1>HooksApp</h1>
        <CounterApp />
        <hr />
        <CounterWithCustomHook />
    </div>
  )
}
