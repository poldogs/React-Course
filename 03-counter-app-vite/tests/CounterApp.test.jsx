import { fireEvent, render, screen } from '@testing-library/react';
import { CounterApp } from '../src/CounterApp';


describe('Pruebas en el <CounterApp />', () => {

    const initialValue = 10;
    
    test('de hacer match con el snapshot', () => { 
        const wrapper = render( <CounterApp value={ initialValue } /> );
        expect( wrapper ).toMatchSnapshot();
     })

    test('el valor inicial debe ser 100', () => { 
        render( <CounterApp value={ 100 } /> );
        expect( screen.getByText( 100 ) ).toBeTruthy();
        //expect( screen.getByRole('heading') ).toHaveTextContent('100');
     })

    test('debe de incrementar con el botón +1', () => {
        
        render( <CounterApp value={ initialValue } /> );
        fireEvent.click( screen.getByText('+1') )
        expect( screen.getByText('11') ).toBeTruthy();

    });

    test('debe de decrementar con el botón -1', () => {
        
        render( <CounterApp value={ initialValue } /> );
        fireEvent.click( screen.getByText('-1') );
        expect( screen.getByText('9') ).toBeTruthy();

    });

    test('debe de funcionar el botón de reset', () => {
        
        render( <CounterApp value={ 355 } /> );
        fireEvent.click( screen.getByText('+1') );
        fireEvent.click( screen.getByText('+1') );
        fireEvent.click( screen.getByText('+1') );

        expect( screen.getByText('358') ).toBeTruthy();

        // fireEvent.click( screen.getByText('Reset') );
        fireEvent.click(screen.getByRole('button', { name: 'btn-reset' }));

        expect( screen.getByText( 355 ) ).toBeTruthy();

    });

    test('deben de funcionar todos los botones', () => { 
        render( <CounterApp value={ initialValue } /> );
        fireEvent.click( screen.getByText('+1') );
        fireEvent.click( screen.getByText('+1') );
        fireEvent.click( screen.getByText('+1') );

        expect( screen.getByText('13') ).toBeTruthy();

        fireEvent.click( screen.getByText('-1') );

        expect( screen.getByText('12') ).toBeTruthy();

        fireEvent.click( screen.getByText('Reset') );

        expect( screen.getByText( initialValue ) ).toBeTruthy();
     })




});