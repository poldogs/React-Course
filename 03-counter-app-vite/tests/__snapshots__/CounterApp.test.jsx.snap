// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <CounterApp /> de hacer match con el snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h1>
        CounterApp
      </h1>
      <h2
        id="counter"
      >
         
        10
         
      </h2>
      <button
        id="btn-add"
      >
         +1 
      </button>
      <button
        id="btn-substact"
      >
         -1 
      </button>
      <button
        aria-label="btn-reset"
      >
         Reset 
      </button>
    </div>
  </body>,
  "container": <div>
    <h1>
      CounterApp
    </h1>
    <h2
      id="counter"
    >
       
      10
       
    </h2>
    <button
      id="btn-add"
    >
       +1 
    </button>
    <button
      id="btn-substact"
    >
       -1 
    </button>
    <button
      aria-label="btn-reset"
    >
       Reset 
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
